lambda_src/lambda_function.py

```diff
@@ -83,38 +97,35 @@ def async_flow_poll(destination: Text, batch_id: Text) -> Dict[Text, Any]:
         return {'statusCode': 202}
 
 
-def sync_flow(event: Any, context: Any = None) -> Dict[Text, Any]:
+def process_batch(
```

Comment:
> if I recall my last feedback, it was that we should not pass in all of event and re-get the keys. instead pass in the variables and move the headers pre-processing outside this function

Passed the below as arguments instead, with the headers being already processed in `sync_flow`.

```
processed_headers: Dict[Text, Any],
write_uri: Text,
batch_id: Text,
req_body: Text,
event_path: Text,
```

> this is really hard to read so it might need another read, but it seems like it's only called once? not a super strong preference, but why not just inline this code?

This suggestion came from Prasanth actually.

Reply:
> This suggestion came from Prasanth actually.

Do you recall the rationale? Usually the point of sub-procedures as an abstraction is to have them be called more than once.

Reply:
It was mostly that the function was less readable and maintainable. I thought it made sense, since the original function is nearly 100 LOC and has a bunch of moving parts.